admin/src/js/controllers/upgrade.js

```diff
@@ -20,21 +24,46 @@ angular.module('controllers').controller('UpgradeCtrl',
     $scope.loading = true;
     $scope.versions = {};
 
-    DB().get('_design/medic')
-      .then(function(ddoc) {
-        $scope.deployInfo = ddoc.deploy_info;
+    var getCurrentDeployment = function() {
+      return DB().get(DEPLOY_DOC_ID)
+      .catch(function(err) {
```

Comment:
Actually I don't really need the if block in the then regardless right, because worse case I'm setting a scoped variable to undefined.

I think for a function this small it's OK, but in general I prefer to keep the 404 handling[1] as close to where the 404 might be generated from as possible, to make it clear what it's catching.

[1] I wish PouchDB had a flag to not treat 404s as errors and just return `null` or something. Depending on your use case 404s aren't _really_ errors.

Reply:
Yeah I can see that. I just don't like having to handle the `deployDoc` is null case. It would never be null if you didn't catch the err first. But it's up to you!